utils/citizen_id.ts

```diff
@@ -45,17 +53,64 @@ const verifySupportToken = (
       )
     );
 
-export const withCitizenIdCheck = <T, S>(
+const checkUserGroups = (
+  userId: NonEmptyString,
+  adminGroup: NonEmptyString,
+  adb2cCreds: IServicePrincipalCreds,
+  cacheTtl: NonNegativeInteger
+) =>
+  fromNullable(memcache.get(userId)).foldL(
+    () =>
+      getGraphRbacManagementClient(adb2cCreds)
+        .chain(client =>
+          tryCatch(
+            () =>
+              client.users.getMemberGroups(userId, {
+                securityEnabledOnly: true
+              }),
+            toError
+          ).chain(response =>
+            array.sequence(taskEither)(
+              response.map(groupId =>
+                tryCatch(() => client.groups.get(groupId), toError).map(
+                  _ => _.displayName
+                )
+              )
+            )
+          )
+        )
+        .chain(groupNames => {
+          memcache.put(userId, groupNames, cacheTtl);
+          return taskEither.of(groupNames.includes(adminGroup));
+        })
+        .mapLeft(_ => ResponseErrorInternal(_.message)),
+    (userGroups: readonly NonEmptyString[]) =>
+      taskEither.of(userGroups.includes(adminGroup))
+  );
+
+export const withCitizenIdCheck = (
+  userId: NonEmptyString,
   citizenId: CitizenID,
   publicRsaCertificate: NonEmptyString,
-  f: (fiscalCode: FiscalCode) => TaskEither<T, S>
+  adb2cCreds: IServicePrincipalCreds,
+  canQueryFiscalCodeGroup: NonEmptyString,
+  cacheTtl: NonNegativeInteger
 ) =>
   // TODO insert group check in case of FiscalCode used by non admin users
```

Comment:
is this TODO comment still relevant?

Reply:
Nope, I'll delete it

Reply:
Fixed in [66e5273](https://github.com/pagopa/io-functions-backoffice/pull/17/commits/66e5273edf6e2851642dc6a5eb99400167bb41f0)